### DISCUSSION TITLE
Typing signatures for queries as parameters.

### DISCUSSION THREAD

**Discussion by vorillaz** [prisma/prisma#9865] [Q&A] [answered]
Hey there, I was wondering if there is any "generic" typing signature for Prisma queries, I have a generic helper function function that is used for error thrown purposes that looks like this:
```typescript
export async function runQuery(query) {
    return query.catch(e => {
        throw e;
    });
}
```

And I used to call the function as:
```typescript
const adminUser = await runQuery(
    prisma.user.create({
      data: {
        email,
      },
      include: {accounts: true},
    }),
  );
```

So I would love to have something like a generic query signature in order to describe the `query` parameter.

**Comment by ryands17** [comment, marked as answer]
@vorillaz 👋 

You could try something like this:

```ts
function runQuery<T extends unknown>(query: T) {
  try {
    return query
  } catch (error) {
    throw error
  }
}

let response = await runQuery(prisma.user.findMany())
console.log(response)
```

**Reply by vorillaz** [reply]
Thanks a lot for this one :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
